collector/lib/SysdigService.h

```diff
@@ -101,6 +106,11 @@ class SysdigService : public Sysdig {
   mutable std::mutex running_mutex_;
   bool running_ = false;
   bool useEbpf_ = false;
+
+  void ServePendingProcessRequests();
+  mutable std::mutex process_requests_mutex_;
+  // [ ( pid, callback ), ( pid, callback ), ... ]
+  std::list<std::pair<uint64_t, std::weak_ptr<std::function<void(threadinfo_map_t::ptr_t)>>>> pending_process_requests_;
```

Comment:
Can a typedef be used here to simplify this?

Reply:
It is not reused much, but 4 levels is a lot, I agree. I introduced a typedef for the callback reference.